models/baseModels/Invoice/Invoice.ts

```diff
@@ -162,16 +172,44 @@ export abstract class Invoice extends Transactional {
     await validateBatch(this);
   }
 
+  async getNewBaseGrandTotal() {
+    const totalLotaltyAmount = await getNewGrandTotal(
+      this.loyaltyProgram as string,
```

Comment:
1. the names `getNewBaseGrandTotal`, `getNewGrandTotal` doesnt accurately describe what action it performs.

2. there is a typo in the variable name `totalLotaltyAmount`